cfme/tests/middleware/test_middleware_deployment.py

```diff
@@ -1,33 +1,48 @@
-from random import sample
-
 import pytest
+from cfme.middleware import get_random_list
 from cfme.middleware.deployment import MiddlewareDeployment
+from cfme.middleware.server import MiddlewareServer
 from utils import testgen
 from utils.version import current_version
 
 pytestmark = [
-    pytest.mark.usefixtures('single_middleware_provider'),
+    pytest.mark.usefixtures('setup_provider'),
```

Comment:
@hhovsepy we have two type of test cases here. one with `provider` and another without `provider`.
* When we go without `provider`, we no need to fetch data from `mgmt`. Because we need to query on multiple `mgmt` systems. So I have removed `mgmt` check in this test. Test happens only between `DB` and `UI`.
* When we go with `provider`, In `UI` we have navigation option to select particular `provider`, In `db` can filter with `provider` name. In `mgmt` nothing to do(we are getting `mgmt` from `provider` object).

When we go with `single_middleware_provider`, displays data only from single `provider`. There is no meaning to perform test through `provider`. We already filtered out other providers data by keeping only one `provider`.